src/docs/MLOS/MLOSUserGuide.adoc

```diff
@@ -166,6 +166,414 @@ E. Click the *Tasks* tab, select an *Preview_Results* task and click on the *Pre
+
+```python
+#%connect(path=PATH_TO/YOUR_CONFIG_FILE.ini)
+```
+
+(for more information about this format please check
+https://docs.python.org/3/library/configparser.html[configParser])
+
+=== Usage
+
+===== Creating a Python task
+
+To create a task, please use the pragma `#%task()` followed by the task implementation script wrote into a notebook block code.
+To use this pragma, at least, a task name has to be provided. Example:
+
+```python
+#%task(name=myTask)
+print('Hello world')
+```
+
+General usage:
+
+```python
+#%task(name=TASK_NAME, [language=SCRIPT_LANGUAGE], [dep=[TASK_NAME1,TASK_NAME2,...]], [generic_info=[(KEY1,VAL1), (KEY2,VALUE2),...]], [export=[VAR_NAME1,VAR_NAME2,...]], [import=[VAR_NAME1,VAR_NAME2,...]], [path=IMPLEMENTATION_FILE_PATH])\n'
+```
+
+As seen in the general usage, users can also provide more information about the task by using the pragma's options:
+
+====== Language
+
+The `language` parameter is needed when the task script is not written in native Python, the default language.
+The handled programming languages are:
+
+* Linux_Bash
+* Windows_Cmd
+* DockerCompose
+* Scalaw
+* Groovy
+* Javascript
+* Jython
+* Python
+* Ruby
+* Perl
+* PowerShell
+* R
+
+Example of usage when the task is written in `Linux_Bash`:
+
+```bash
+#%task(name=myTask, language=Linux_Bash)
+echo 'Hello, World!'
+```
+
+====== Dependencies
+
+One of the most important notions in workflows is the dependencies between tasks. To specify this information, please use the
+`dep` parameter. The value should be a list of all the tasks on which the new task depends. Example:
+
+```python
+#%task(name=myTask,dep=[parentTask1,parentTask2])
+print('Hello world')
+```
+
+====== Generic information
+
+To specify the values of the advanced ProActive variables
+https://doc.activeeon.com/latest/user/ProActiveUserGuide.html#_generic_information[generic_information],
+ the parameter `generic_info` is provided. The value should be a tuples `(key,value)` list of all the names and values of the
+ProActive parameters. Example:
+
+```python
+#%task(name=myTask, generic_info=[(var1,value1),(var2,value2)])
+print('Hello world')
+```
+
+====== Export/import variables
+
+The `export` and `import` parameters make possible variables propagation between the different tasks of a workflow.
+If `myTask1` variables `var1` and `var2` are needed in `myTask2`, the `myTask1` pragma should include and `export` with
+a list of these variable names and the `myTask2` pragma an `import` with a list including the same names. Example:
+
+`myTask1` implementation bloc would be:
+```python
+#%task(name=myTask1, export=[var1,var2])
+var1 = "Hello"
+var2 = "ActiveEon!"
+```
+
+and `myTask2` implementation bloc would be:
+```python
+#%task(name=myTask2, dep=[myTask1], import[var1,var2])
+print(var1 + " from " + var2)
+```
+
+====== Implementation file
+
+It is possible to use an external implementation file as task implementation. To do so, the option `path` should be used.
```

Comment:
I'm not sure if you call it option or parameter, but use only one name to call it, I think I saw the word parameter above)

Reply:
I used option here to underline the fact that the parameter is optional.

Reply:
It is (also?) possible to use an external file to define the task implementation.